app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,56 @@
+class CommentsController < ApplicationController
+  before_action :find_post
+  before_action :find_comment, only: [:edit, :update, :destroy]
+
+  def create
+    @comment = @post.comments.new comment_params
+    if @comment.save
+      respond_to do |format|  
+        format.html { redirect_to @comment.post, notice: 'Comment was successfully created.' }
+        format.js { render layout: false }
+        format.json { render @comment.post, status: :created, location: @comment }
+      end
+    else 
+      respond_to do |format|  
+        format.html { redirect_to @comment.post, notice: 'Please sign or sign up to comment' }
```

Comment:
It's also need `format.js`, `format.json`. Should have same response for formats like above code